madgraph/interface/reweight_interface.py

```diff
@@ -708,6 +760,235 @@ def do_launch(self, line):
         
         self.options['rwgt_name'] = None
 
+    def set_initial_mass_to_zero(self, event):
+
+        event[0].mass = 0.
+        event[1].mass = 0.
+        tot_E=0.
+        for ip,part in enumerate(event):
+            if ip >=2 :
+                tot_E += part.E
+        if (event[0].pz > 0.and event[1].pz < 0):
+            event[0].set_momentum(lhe_parser.FourMomentum([tot_E/2., event[0].px, event[0].py, tot_E/2.]))
+            event[1].set_momentum(lhe_parser.FourMomentum([tot_E/2., event[0].px, event[0].py, -tot_E/2.]))
+        elif (event[0].pz < 0.and event[1].pz > 0):
+            event[0].set_momentum(lhe_parser.FourMomentum([tot_E/2., event[0].px, event[0].py, -tot_E/2.]))
+            event[1].set_momentum(lhe_parser.FourMomentum([tot_E/2., event[0].px, event[0].py, tot_E/2.]))
+        else:
+            logger.critical('ERROR: two incoming partons not back.-to-back')
+
+    def set_final_jet_mass_to_zero(self, event):
+
+        for ip,part in enumerate(event):
+            if ((abs(part.pid) <= 5) or (abs(part.pid) == 11) or (abs(part.pid) == 12)) and (ip > 1):
+                event[ip].mass = 0.
+                E_1_new = math.sqrt(event[ip].mass**2 + event[ip].px**2 + event[ip].py**2 + event[ip].pz**2)
+                event[ip].set_momentum(lhe_parser.FourMomentum([E_1_new, event[ip].px, event[ip].py, event[ip].pz]))
```

Comment:
```suggestion
                part.mass = 0.
                E_1_new = math.sqrt(part.mass**2 + part.px**2 + part.py**2 + part.pz**2)
                part.set_momentum(lhe_parser.FourMomentum([E_1_new, part.px, part.py, part.pz]))
```
just to be more consistent 
Note if you move those lines to lhe_parser note that you need to remove the lhe_parser.